lib/ClangImporter/SwiftDeclSynthesizer.cpp

```diff
@@ -2534,128 +2534,167 @@ SwiftDeclSynthesizer::makeDefaultArgument(const clang::ParmVarDecl *param,
+    for (const auto *param : selectedCtorDecl->parameters())
+      paramTypes.push_back(param->getType());
+    clang::FunctionProtoType::ExtProtoInfo EPI;
+    clang::QualType funcTypeToSynthesize =
+        clangCtx.getFunctionType(cxxRecordPtrTy, paramTypes, EPI);
+
+    clang::CXXMethodDecl *synthesizedCxxMethodDecl =
+        clang::CXXMethodDecl::Create(
+            clangCtx, const_cast<clang::CXXRecordDecl *>(cxxRecordDecl),
+            cxxRecordDeclLoc,
+            clang::DeclarationNameInfo(funcNameToSynthesize, cxxRecordDeclLoc),
+            funcTypeToSynthesize,
+            clangCtx.getTrivialTypeSourceInfo(funcTypeToSynthesize),
+            clang::SC_Static, /*UsesFPIntrin=*/false, /*isInline=*/true,
+            clang::ConstexprSpecKind::Unspecified, cxxRecordDeclLoc);
+    assert(
+        synthesizedCxxMethodDecl &&
+        "Unable to synthesize static factory for c++ foreign reference type");
+    synthesizedCxxMethodDecl->setAccess(clang::AccessSpecifier::AS_public);
+
+    llvm::SmallVector<clang::ParmVarDecl *, 4> synthesizedParams;
+    for (unsigned int i = 0; i < ctorParamCount; ++i) {
+      auto *origParam = selectedCtorDecl->getParamDecl(i);
+      auto *param = clang::ParmVarDecl::Create(
+          clangCtx, synthesizedCxxMethodDecl, cxxRecordDeclLoc,
+          cxxRecordDeclLoc, origParam->getIdentifier(), origParam->getType(),
+          clangCtx.getTrivialTypeSourceInfo(origParam->getType()),
+          clang::SC_None, nullptr);
+      synthesizedParams.push_back(param);
+    }
+    synthesizedCxxMethodDecl->setParams(synthesizedParams);
+
+    if (!hasImmortalAttrs(cxxRecordDecl)) {
+      clang::SwiftAttrAttr *returnsRetainedAttrForSynthesizedCxxMethodDecl =
+          clang::SwiftAttrAttr::Create(clangCtx, "returns_retained");
+      synthesizedCxxMethodDecl->addAttr(
+          returnsRetainedAttrForSynthesizedCxxMethodDecl);
+    }
+
+    std::string swiftInitStr = "init(";
+    for (unsigned i = 0; i < ctorParamCount; ++i) {
+      swiftInitStr += "_:";
+    }
+    swiftInitStr += ")";
+    clang::SwiftNameAttr *swiftNameAttr =
+        clang::SwiftNameAttr::Create(clangCtx, swiftInitStr);
+    synthesizedCxxMethodDecl->addAttr(swiftNameAttr);
+
+    llvm::SmallVector<clang::Expr *, 8> ctorArgs;
+    for (auto *param : synthesizedParams) {
+      ctorArgs.push_back(clang::DeclRefExpr::Create(
+          clangCtx, clang::NestedNameSpecifierLoc(), clang::SourceLocation(),
+          param,
+          /*RefersToEnclosingVariableOrCapture=*/false, clang::SourceLocation(),
+          param->getType(), clang::VK_LValue));
+    }
+    llvm::SmallVector<clang::Expr *, 8> ctorArgsToAdd;
+    if (clangSema.CompleteConstructorCall(selectedCtorDecl, cxxRecordTy,
+                                          ctorArgs, cxxRecordDeclLoc,
+                                          ctorArgsToAdd))
+      continue;
+
+    clang::ExprResult synthesizedConstructExprResult =
```

Comment:
Nit: The whole function is about synthesizing declarations and expressions. I think we can drop the "synthesized" prefix from these identifier names. It is clear from the context that we are creating them.